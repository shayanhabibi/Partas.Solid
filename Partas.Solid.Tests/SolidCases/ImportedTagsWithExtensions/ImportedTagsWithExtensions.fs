module Partas.Solid.Tests.SolidCases.ImportedTagsWithExtensions

open Partas.Solid.Tests.Shared.ImportedTagsWithExtensions
open Partas.Solid
open Fable.Core

type [<Erase>] NoPropsOneExtension() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        Imported().attr("key", "value")

type [<Erase>] PropsWithExtension() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        CustomTag(class' = "testClass").attr("key", "value")
        
type [<Erase>] NoPropsExtensionChain() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        CustomTag().attr("key", "value").attr("key2", "value2")

type [<Erase>] PropsWithExtensionChain() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.typeDef =
        Imported(class' = "Tester", draggable = "drag").attr("key", "value").attr("key2", "value2").attr("key3", "value3").attr("key4", "value4")
