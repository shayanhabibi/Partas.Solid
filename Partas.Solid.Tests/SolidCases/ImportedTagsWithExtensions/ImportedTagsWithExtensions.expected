import { splitProps } from "solid-js";
import { FakeImport } from "FakeLibrary";
import { CustomTag } from "./Shared.fs.jsx";

export function NoPropsOneExtension(props) {
    const [local, others] = splitProps(props, []);
    return <FakeImport key="value" />;
}

export function PropsWithExtension(props) {
    const [local, others] = splitProps(props, []);
    return <CustomTag class="testClass"
        key="value" />;
}

export function NoPropsExtensionChain(props) {
    const [local, others] = splitProps(props, []);
    return <CustomTag key="value"
        key2="value2" />;
}

export function PropsWithExtensionChain(props) {
    const [local, others] = splitProps(props, []);
    return <FakeImport class="Tester"
        draggable="drag"
        key="value"
        key2="value2"
        key3="value3"
        key4="value4" />;
}

